src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -165,8 +165,14 @@ public AzureAppConfigurationOptions Select(string keyFilter, string labelFilter
             {
                 throw new ArgumentException("The characters '*' and ',' are not supported in label filters.", nameof(labelFilter));
             }
-
-            if (!_kvSelectors.Any(s => string.Equals(s.KeyFilter, keyFilter) && string.Equals(s.LabelFilter, labelFilter)))
+            var foundExactSameSelector = _kvSelectors.Find(s => string.Equals(s.KeyFilter, keyFilter) && string.Equals(s.LabelFilter, labelFilter));
```

Comment:
We generally use `FirstOrDefault` instead of `Find`

Reply:
Searching the web a little bit, looks like `Find` has better performance (though a little bit). 
About the usage, here both methods return null if not found (if I don't specify a default value in FirstOrDefault). Just curious about the benefit of using `FirstOrDefault` here? Consistency?

Reply:
Consistency, yes. Also, Find is only implemented for List type. FirstOrDefault is more generic for IEnumerable type.